contrib/IECoreAlembic/include/IECoreAlembic/AlembicScene.h

```diff
@@ -0,0 +1,124 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2017, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef IECOREALEMBIC_ALEMBICSCENE_H
+#define IECOREALEMBIC_ALEMBICSCENE_H
+
+#include "IECore/SampledSceneInterface.h"
+
+#include "IECoreAlembic/Export.h"
+#include "IECoreAlembic/TypeIds.h"
+
+namespace IECoreAlembic
+{
+
+class IECOREALEMBIC_API AlembicScene : public IECore::SampledSceneInterface
+{
+	public :
+
+		IE_CORE_DECLARERUNTIMETYPEDEXTENSION( AlembicScene, IECoreAlembic::AlembicSceneTypeId, IECore::SampledSceneInterface );
+
+		AlembicScene( const std::string &fileName, IECore::IndexedIO::OpenMode mode );
```

Comment:
Yep, it does for sure. But the base SceneInterface class uses IndexedIO for `create()` and `supportedExtensions()` so it has to be used here really. Would be easy enough just to define the equivalent enum on the SceneInterface directly though - shall I do that in a subsequent PR?